app/src/androidTest/java/com/karhoo/karhootraveller/splash/SplashTestsEndToEnd.kt

```diff
@@ -0,0 +1,207 @@
+    @Test
+    @AllowFlaky(attempts = 5)
+    fun userLogsInAndCompletesATrip() {
+        serverRobot {
+            successfulToken()
+            sdkInitResponse(HTTP_OK, BRAINTREE_TOKEN)
+            userProfileResponse(HTTP_OK, USER_INFO)
+            paymentsProviderResponse(HTTP_OK, ServerRobot.BRAINTREE_PROVIDER)
+            paymentsNonceResponse(HTTP_OK, ServerRobot.PAYMENTS_TOKEN)
+            reverseGeocodeResponse(HTTP_OK, REVERSE_GEO_SUCCESS)
+        }
+        splash(this) {
+            sleep(MEDIUM)
+            clickOnSignInButton()
+        }
+        login {
+            validEmailAndPasswordLogin()
+            userClicksOnSignInButton()
+        }
+        booking {
+            sleep(MEDIUM)
+        } result {
+            checkBookingScreenIsShown()
+        }
+        serverRobot {
+            addressListResponse(HTTP_OK, PLACE_SEARCH_RESULT)
+            addressDetails(HTTP_OK, PLACE_DETAILS)
+        }
+        booking {
+            clickPickUpAddressField()
+        }
+        address {
+            search(SEARCH_ADDRESS)
+            clickBakerStreetResult()
+        }
+        booking {
+            sleep(MEDIUM)
+        } result {
+            selectedPickupAddressIsVisible(address = SELECTED_ADDRESS)
+        }
+        serverRobot {
+            addressListResponse(HTTP_OK, PLACE_SEARCH_RESULT_EXTRA)
+            addressDetails(HTTP_OK, PLACE_DETAILS_EXTRA)
+        }
+        booking {
+            clickDestinationAddressField()
+        }
+        address {
+            search(SEARCH_ADDRESS_EXTRA)
+            sleep(LONG)
+            clickOxfordStreetResult()
+        }
+        serverRobot {
+            quoteIdResponse(HTTP_OK, ServerRobot.QUOTE_LIST_ID_ASAP)
+            quotesResponse(HTTP_OK, ServerRobot.VEHICLES_V2_ASAP)
+        }
+        booking {
+            sleep(MEDIUM)
+        } result {
+            selectedDestinationAddressIsVisible(address = SELECTED_ADDRESS_EXTRA)
+        }
+        serverRobot {
+            bookingResponse(HTTP_OK, TestData.TRIP)
+            bookingStatusResponse(code = HTTP_OK, response = ServerRobot.TRIP_STATUS_DER, trip = TestData.TRIP.tripId)
+            driverTrackingResponse(code = HTTP_OK, response = ServerRobot.DRIVER_TRACKING, trip = TestData.TRIP.tripId)
+            bookingDetailsResponse(code = HTTP_OK, response = ServerRobot.TRIP_DER_NO_NUMBER_PLATE, trip = TestData.TRIP.tripId)
+        }
+        booking {
+            waitForTime(7000)
+            pressFirstQuote()
+            pressBookRideButton()
+        } result {
+            checkDriverDetails()
+        }
+    }
+
+    companion object {
+
+        private val origin = TestData.ORIGIN_TRIP
+        private val destination = TestData.DESTINATION_TRIP
+        private val initialTripDetails = TripInfo(origin = origin, destination = destination)
+        private val identicalAddresses = TripInfo(origin = origin, destination = origin)
+
+//        val CLEAN_TRIP_INTENT = Intent().apply {
```

Comment:
Is this needed later or can it be deleted?

Reply:
Oh sugar, I forgot to take it out. I'll do that now

Reply:
It's from my other branch and not stashing